src/coins.cpp

```diff
@@ -443,6 +443,18 @@ double CCoinsViewCache::GetPriority(const CTransaction &tx, int nHeight) const
             dResult += coins->vout[txin.prevout.n].nValue * (nHeight-coins->nHeight);
         }
     }
+
+    // Joinsplits do not reveal any information about a note's amount or age.
+    // We need to ensure priority increases the longer a tx remains in the mempool, so we don't
+    // generate random values which would change on each invocation of this method.
+    // Instead, we use the anchor as a seed to generate the note value deterministically.
+    // Currently, we say that each JoinSplit has input notes of between 1 to 2 ZEC,
+    // and that the age of each note is half the age of the blockchain.
+    for (auto & jsdesc : tx.vjoinsplit) {
+        uint64_t i = jsdesc.anchor.GetCheapHash();
```

Comment:
Technically this means that miners can control the priorities of pure-JoinSplit transactions in blocks after theirs, since they can add or remove transactions from their block to influence the resulting anchor. But with the floor of 1 ZEC, I doubt this would result in a useful attack.